observers/qos_correction.cpp

```diff
@@ -0,0 +1,200 @@
+#include <list>
+#include <vector>
+
+#include "observers/qos_correction.hpp"
+
+#include "mesos/slave/oversubscription.hpp"
+
+#include "messages/serenity.hpp"
+
+#include "serenity/wid.hpp"
+
+namespace mesos {
+namespace serenity {
+
+using std::list;
+
+/**
+ * Checks contentions and choose executors to kill.
+ * Currently we have sorted Contentions by severity.
+ * We are iterating from the most serious contention to the less serious.
+ * WARNING(bplotka): This interpreter assumes that severity is reflecting
```

Comment:
What is the best word for the entity which takes contentions and create corrections based on it? Decider?